src/core/or/circuitpadding.c

```diff
@@ -442,7 +442,7 @@ circpad_machine_sample_delay(circpad_machine_runtime_t *mi)
     return circpad_distribution_sample_iat_delay(state, iat_delay_shift);
   } else if (state->token_removal != CIRCPAD_TOKEN_REMOVAL_NONE) {
     /* We have a mutable histogram. Do basic sanity check and apply: */
-    if (BUG(!mi->histogram) ||
+    if (BUG(!CIRCPAD_IS_TOKEN_REMOVAL_SUPPORTED(mi)) ||
```

Comment:
This is not exactly the same as it was before -- histogram_len was not previously checked.  I'm assuming that's okay.

Reply:
Yes, this was why I introduced this macro, to ensure that we have equivalence between histogram == NULL, and histrogram_length == 0, and also state_t.token_removal == CIRCPAD_TOKEN_REMOVAL_NONE. All those should be set at the same time.

When I transform this into a static inline as per your other comment, I will make that function check all of the things. (The state check is more expensive since we don't always have a direct pointer into the state, and it requires several dereferences to get one).